lib/internal/errors.js

```diff
@@ -40,9 +40,42 @@ const prepareStackTrace = (globalThis, error, trace) => {
   if (trace.length === 0) {
     return errorString;
   }
-  return `${errorString}\n    at ${trace.join('\n    at ')}`;
+
+  const { getOptionValue } = require('internal/options');
+  if (getOptionValue('--experimental-source-maps')) {
```

Comment:
I am for caching `getOptionValue` but I am not sure I agree this is a hot path. Errors are an exceptional path and we have always treated them as something relatively rare where we want to optimize for user experience rather than speed.

Reply:
well as long as we can make them fast, i see no reason not to :smile:

Reply:
I'm more e worried about the check in maybeCacheSoureMap, which is a hot path; I'll cache the value in both places 👌